src/action/macos/create_determinate_nix_volume.rs

```diff
@@ -72,14 +73,28 @@ impl CreateDeterminateNixVolume {
 
         let create_synthetic_objects = CreateSyntheticObjects::plan().await.map_err(Self::error)?;
 
-        let unmount_volume = UnmountApfsVolume::plan(disk, name.clone())
-            .await
-            .map_err(Self::error)?;
-
         let create_volume = CreateApfsVolume::plan(disk, name.clone(), case_sensitive)
             .await
             .map_err(Self::error)?;
 
+        let unmount_volume = {
+            let mut task = UnmountApfsVolume::plan(disk, name.clone())
+                .await
+                .map_err(Self::error)?;
+
+            if create_volume.state == crate::action::ActionState::Completed {
+                if let Ok(diskinfo) = DiskUtilInfoOutput::for_volume_name(&name).await {
+                    if Path::new(&diskinfo.parent_whole_disk) == disk
+                        && diskinfo.mount_point.as_deref() == Some(Path::new("/nix"))
+                    {
+                        task.state = crate::action::ActionState::Skipped
+                    }
+                }
+            }
+
+            task
```

Comment:
We discussed moving this into another `plan` function on `UnmountApfsVolume` (i.e. `UnmountApfsVolume::plan_volume_already_existed`) that controls this instead of this branching.